### PR TITLE
helpers: Remove extra '-' from MakePath (Fix for issue 2788) 

### PR DESCRIPTION
**Pull request by bryanjeal**
This Pull Request is a fix for issue #2788. 

MakePath function would leave extra '-' characters in its output when Unicode Characters were removed after strings.Replace is run.

### CODE REVIEW

**Comment by bep** [comment]
You need to run `go fmt` on your code to get the CI to go green.

**Comment by bryanjeal** [comment]
Sorry about that @bep. 

I fixed a conflict in github's editor and that messed with the formatting. Should be fixed now.

**Comment by moorereason** [comment]
@bryanjeal, please sign the CLA.  Otherwise, you can close this PR.  Thanks.

File: helpers/path.go [hunk 1]
```diff
@@ -124,7 +127,7 @@ func (p *PathSpec) UnicodeSanitize(s string) string {
 	for i, r := range source {
 		if r == '%' && i+2 < len(source) && ishex(source[i+1]) && ishex(source[i+2]) {
 			target = append(target, r)
-		} else if unicode.IsLetter(r) || unicode.IsDigit(r) || unicode.IsMark(r) || r == '.' || r == '/' || r == '\\' || r == '_' || r == '-' || r == '#' || r == '+' || r == '~' {
+		} else if unicode.IsLetter(r) || unicode.IsDigit(r) || unicode.IsMark(r) || r == '.' || r == '/' || r == '\\' || r == '_' || r == '-' || r == '#' || r == '+' || r == '~' || r == ' ' {
```
Comment (bep) [inline helpers/path.go]: This should be superflous if you adjust the order of the above.

**Comment by bep** [comment]
What is the performance implications of this change?

File: helpers/path.go (hunk 1 above)
Comment (bryanjeal) [reply helpers/path.go]: Please excuse my ignorance, but I can't seem to reorder the operations in the `MakePath` function to make adding the space to the `UnicodeSanitize` unnecessary.

Changing to: Regex ReplaceAll -> UnicodeSanitize -> Trim 
causes the test case: `{"Foo & bar", "Foo-bar", true}` to fail.

**Comment by bryanjeal** [comment]
@bep please find the benchmark results below:

### Code

```
func BenchmarkMakePath(b *testing.B) {
	v := viper.New()
	l := NewDefaultLanguage(v)
	v.Set("removePathAccents", true)
	p, _ := NewPathSpec(hugofs.NewMem(v), l)

	for i := 0; i < b.N; i++ {
		_ = p.MakePath("Foo & bar")
	}
}
```

### Results

Results based on the code currently in `gohugoio/hugo` **master**:
```bash
$ go test -bench=MakePath
BenchmarkMakePath-8   	  500000	      2976 ns/op
PASS
```

Results based on the code in this change request:
```bash
$ go test -bench=MakePath
BenchmarkMakePath-8   	  300000	      3621 ns/op
PASS
```

**Pull request closed without merging**